File: map-service/src/main/java/com/example/map/controller/RestaurantSearchController.java
package com.example.map.controller;

import com.example.common.dto.ApiResponse;
import com.example.map.dto.RestaurantSearchResponse;
import com.example.map.service.RestaurantSearchService;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RestController
@RequestMapping("/api/map/restaurants")
@RequiredArgsConstructor
public class RestaurantSearchController {

    private final RestaurantSearchService restaurantSearchService;

    @GetMapping("/search")
    public ApiResponse<List<RestaurantSearchResponse>> search(
            @RequestParam String keyword,
            @RequestParam Double x,
            @RequestParam Double y,
            @RequestParam(defaultValue = "1000") Integer radius) {
        List<RestaurantSearchResponse> results = restaurantSearchService.search(keyword, x, y, radius);
        return ApiResponse.success("식당 검색 성공", results);
    }
}

File: apps/map-service/src/restaurant/restaurant-search.controller.ts
import { Controller, Get, Query } from '@nestjs/common';
import { ApiResponse } from '@app/common';
import { RestaurantSearchService } from './restaurant-search.service';
import { RestaurantSearchResponseDto } from './dto/restaurant-search-response.dto';

@Controller('api/map/restaurants')
export class RestaurantSearchController {
  constructor(private readonly restaurantSearchService: RestaurantSearchService) {}

  @Get('search')
  async search(
    @Query('keyword') keyword: string,
    @Query('x') x: number,
    @Query('y') y: number,
    @Query('radius') radius: number = 1000,
  ): Promise<ApiResponse<RestaurantSearchResponseDto[]>> {
    const results = await this.restaurantSearchService.search(keyword, x, y, radius);
    return ApiResponse.success('식당 검색 성공', results);
  }
}
